From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shane Freeder <theboyetronic@gmail.com>
Date: Wed, 3 Nov 2021 20:22:36 -0400
Subject: [PATCH] Cleanup allocated favicon ByteBuf

Cleanups a bytebuffer which was allocated during the encoding of the
favicon to be sent to the client.

diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index bfaa8761e925e15cd4f641b95553252294271d5f..e9fd0da6700901cdf794ecc7edf3666e799c555f 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -751,6 +751,7 @@ public abstract class MinecraftServer extends com.hpfxd.pandaspigot.tickloop.Ree
 
         if (file.isFile()) {
             ByteBuf bytebuf = Unpooled.buffer();
+            ByteBuf bytebuf1 = null; // PandaSpigot - cleanup favicon bytebuf
 
             try {
                 BufferedImage bufferedimage = ImageIO.read(file);
@@ -758,13 +759,14 @@ public abstract class MinecraftServer extends com.hpfxd.pandaspigot.tickloop.Ree
                 Validate.validState(bufferedimage.getWidth() == 64, "Must be 64 pixels wide", new Object[0]);
                 Validate.validState(bufferedimage.getHeight() == 64, "Must be 64 pixels high", new Object[0]);
                 ImageIO.write(bufferedimage, "PNG", new ByteBufOutputStream(bytebuf));
-                ByteBuf bytebuf1 = Base64.encode(bytebuf);
+                /*ByteBuf */bytebuf1 = Base64.encode(bytebuf); // PandaSpigot - cleanup favicon bytebuf
 
                 serverping.setFavicon("data:image/png;base64," + bytebuf1.toString(Charsets.UTF_8));
             } catch (Exception exception) {
                 MinecraftServer.LOGGER.error("Couldn\'t load server icon", exception);
             } finally {
                 bytebuf.release();
+                if (bytebuf1 != null) bytebuf1.release(); // PandaSpigot - cleanup favicon bytebuf
             }
         }
 
